client/src/components/utilities.ts

```diff
@@ -11,12 +11,27 @@ export const tryJson = (responseText: string) => {
   }
 }
 
+const parseCookies = () =>
+  Object.fromEntries(
+    document.cookie.split(';').map(pair => pair.trim().split('='))
```

Comment:
hmmm, so this is not as secure as it could be.

The most secure configuration would be to make the cookie `httpOnly`, which then means the cookie is not accessible from JavaScript, which is an XSS defense.

The CSRF token is then delivered via a call to `/csrf` or the like, that explicitly hands the CSRF cookie to the JS client.

Reply:
Based on the [docs](https://flask-seasurf.readthedocs.io/en/latest/#ajax-usage) I have found for the library, it seems like reading the token from the cookie is the standard way to do it. It does have one [recommendation](https://docs.djangoproject.com/en/dev/ref/csrf/#acquiring-the-token-if-csrf-use-sessions-or-csrf-cookie-httponly-is-true) for when the cookie is httpOnly, which is to "...include the CSRF token in your HTML and read the token from the DOM with JavaScript." Does that sound better than reading it from the cookie to you?

I also saw this discussion on StackExchange where many people think that CSRF token cookies don't require httpOnly: https://security.stackexchange.com/questions/175536/does-a-csrf-cookie-need-to-be-httponly.

Curious to discuss

Reply:
@jonahkagan after some thought and consulting with websec colleagues, I concur with your conclusion: reading `document.cookie` is best.

One thing you're doing particularly well which I want to make sure we don't lose in some future optimization: you're reading the CSRF cookie every time you make a request. That's the right thing to do. Let's not lose that by caching it in local state or something like that. (Reason: CSRF token can change for a number of reasons. Best to re-read it and use the freshest every time.)